### DISCUSSION TITLE
SmartGrid Schalten

### DISCUSSION THREAD

**Discussion by MichaelGer1410** [Trunks1982/Daikin-Rotex-HPSU-CAN#40] [General]
Hallo,

vorab: Habe alles mit kabelbrücken ersteinmal zusammengesteckt, angeschlossen und siehe da es hat auf anhieb funktioniert. Sofort wurden sämtliche daten ausgelesen!
Könntest du evtl. vielleicht noch SmartGrid implementieren so das man dies ebenfalls ein bzw ausschalten kann?

**Comment by Trunks1982** [comment]
Ganz ehrlich flexibler bist du wenn du Ww so erhöhst mit einer Automatisiserung. Sg ist nicht so der hit und springt auch nicht immer an.Ich benutze Sg auch nicht mehr und erhöhe einfach ww oder Raumsoll

**Reply by magic2910** [reply]
Auf wieviel Grad ww und raumsoll gehst du? Ich teste aktuell mit 60 grad ww

**Reply by Trunks1982** [reply]
54Grad.reicht für uns.

**Reply by magic2910** [reply]
Für was verwendet ihr den Button "Warmwasser bereiten" ?

**Reply by Trunks1982** [reply]
Ich zb wenn ich gleich also ca 30min ww haben will.Gehe dann nach ner halben std zb duschen und er heizt es bist soll auf.

**Reply by magic2910** [reply]
Was habt ihr da als Automation im HomeAssistant? Wenn soviel Watt in Überproduktion, dann Warmawasser auf 54 Grad?
Hättet ihr da mal ein Beispiel, ich steht da grad etwas auf dem Schlauch, was man alles beachten sollte.
Danke :)

**Reply by Trunks1982** [reply]
Ja genau wenn überschuss länger als 10min ist soll ww auf 54 grad erhitzt werden.

**Comment by MichaelGer1410** [comment]
Wenn meine PV Überschuss macht gibt die Rotex GO. Das würde ich gerne nutzen und auch abschalten können. 
Zu einer Automatisierung habe ich mir noch keine Gedanken gemacht. Habe noch FHEM, ioBroker ist relativ frisch und damit kenne ich mich noch nicht so aus.

**Reply by Trunks1982** [reply]
Kann ich einbauen brauchst aber dann nen Relais was geschaltet wird.Und dann kannst auch einen Shelly nehmen.Wie gesagt alles schon getestet und am  besten ist eine Automatisiserung.

**Reply by MichaelGer1410** [reply]
Wäre super wenn du das noch einbauen könntest :) 
Wofür ist das Relais? Bisher ist es so das wenn die PV ein Relais schaltet und auf GO schaltet, die Heizung fängt dann bei überschuss an WW zu machen, jenachdem. Verbunden ist die Heizung schon mit der SG Schnitstelle...
Oder habe ich da einen Denkfehler wenn der ESP noch dazu kommt und schaltet? Ich denke das ist ja genau das gleiche als wenn ich am Heizungspanel die Funktion ein oder Ausschalte oder nicht?  
So ganz ist mir auch noch nicht klar wie ich die Schalter für Heizen, Sommer etc. in VIS als schalter einbauen soll.

::EDIT::
Weisst du vielleicht fürfür die Funktion SG-Modus ist?

**Reply by Trunks1982** [reply]
Die gpios müssen ein relais schalten.Wenn man die direkt an sg anschliesst funktioniert das für ein paar sek und es schaltet wieder ab.Deswegen ist nen shelly 1 dafür am besten geeignet.Nach nem extra Relais für nen Esp hab ich noch nie geguckt.
Gleiche ist für Heizen und Kühlen.

**Reply by Trunks1982** [reply]
Mit VIS kann ich dir nicht helfen hab mein über Homeassit Visualisiert.Ich hab zb über ein Drop Down Menu gemacht.

**Reply by MichaelGer1410** [reply]
"> _Die gpios müssen ein relais schalten_.Wenn man die direkt an sg anschliesst funktioniert das für ein paar sek und es schaltet wieder ab.Deswegen ist nen shelly 1 dafür am besten geeignet.Nach nem extra Relais für nen Esp hab ich noch nie geguckt. Gleiche ist für Heizen und Kühlen"

Meine gedanke ist einfach, das man die Funktion im Menü der Anlage einfach aktivieren/deaktivieren kann so wie du es mit den betriebsarten gelöst hast.

**Reply by Trunks1982** [reply]
Aso ja das kann ich einfügen.

**Reply by MichaelGer1410** [reply]
Das wäre toll! 
Habe mir heute mal HA auf eine VM gepackt. Muss mich ersteinmal damit ein wenig beschäftigen...

**Reply by Trunks1982** [reply]
🤣👍

**Reply by MichaelGer1410** [reply]
Guten Morgen, wollte mal vorsichtig anfragen ob du schon dabei bist SG einzubauen? 🙈

**Reply by Trunks1982** [reply]
Bis jetzt noch nicht.Vielleicht heute.

**Reply by MichaelGer1410** [reply]
Das wäre cool

**Reply by Trunks1982** [reply]
So ist implementiert.

**Reply by MichaelGer1410** [reply]
Super vielen Dank! Kann ich jetzt die Rotex.bin über OTA updaten?

**Reply by Trunks1982** [reply]
Ja kannst.

**Comment by ap20132022** [comment]
Ich glaube, dass die aktuelle SG Funktion nicht wirklich korrekt. Es mag sein, dass es für @MichaelGer1410 wunderbar funktioniert. Bisher hab' ich die Funktion SG aus ESPAltherma verwendet und es ist auch in der Bedienanleitung von Rotex beschrieben.
Zusammengefasst: man braucht dafür 2 Ausgänge (2 Pins aus ESP)
0: 00: Free running: WP läuft normal
1: 01: Forced off : WP schaltet aus (wie wenn der Netzbetreiber das Signal zu WP schickt)
2: 10: Recommend on: HP will increase DHW setpoint as well as LW setpoint (documentation says +5 °C, but my tests actually show +6 °C)
3: 11: Forced on: HP will increase DHW setpoint and will use its full power to heat DHW (to 70 °C)
Verzeihung, bin zu faul zu übersetzen.

**Reply by Trunks1982** [reply]
Welche Funktion?
Ich habe nur die Verstellung implementiert.
Die eigentlich Schalter wie ich es mal vor hatte sind nicht vorhanden.

**Reply by MichaelGer1410** [reply]
Damit schalte ich nur ein oder aus. Der rest ist Verkabelung (bei 2 drähten kann man nicht viel falsch machen). Und das Rundsteuergät macht da auch Probleme. Das musste ich vom Netzbetreiber ausbauen lassen das sonst nur SG 3 läuft und immer auf 70 Grad aufheizen bei PV Überschuss ist auch verschnekt.

**Reply by ap20132022** [reply]
> Welche Funktion? Ich habe nur die Verstellung implementiert. Die eigentlich Schalter wie ich es mal vor hatte sind nicht vorhanden.

jetzt verstehe ich. Dachte, dass es schon implementiert wurde. Vielleicht kommt es noch?

**Reply by Trunks1982** [reply]
Hast du es in espaltherma benutzt?
Ging das ganz normal nur über die gpios?

**Reply by ap20132022** [reply]
> Hast du es in espaltherma benutzt? Ging das ganz normal nur über die gpios?

in der Vergangenheit hab' ich es oft benutzt. Seit dem ich dein Projekt verwende, nutze ich SG von ESPaltherma nur selten. Man kann auch über Temperatur gut regeln. Es ist m.M.n "nice to have".
(klar, man muss noch das Signal vom Rundsteuerempfänger trennen).
Es ging über die GPIOs. Bin die Meinung, dass es für dich kein Hexenwerk ist. Es ist "nur" ein Dropdown Menü und anhand der Beschreibung schaltet man die 2 GPIOs ein und aus. 
Kenne noch zu wenig mit ESPhome. Hab' schon angefangen, einzuarbeiten. Wahrscheinlich braucht man ein Labda und switch Funktion.

**Reply by Trunks1982** [reply]
Ich habs mal mit nem switch getestet aber irgendwie gings immer wieder aus.Müsste ich nochmal testen.

**Reply by ap20132022** [reply]
Vorab: hab einen Punkt noch vergessen: ich verwende die GPIOs + Koppelrelais.

Als ich das erste Mal die Kühlungsfunktion aus deinem Projekt verwende, habe ich auch das Problem gehabt. Schuld war schlechte WLAN Verbindung und Firmware Update. 
Hab sogar überlegt, ob ich es negiere soll. 
Mein Koppelrelais kann sowohl als "Open" und als "Close" arbeiten.
Es hat immer Vor- und Nachteile. Aktuell lasse ich original wie es ist. Im Winter, teste ich es nochmal

**Reply by Trunks1982** [reply]
Ah ok du benutzt nen relais.Ich glaub einfacher ist es dann zb ein shelly zu nutzen.Oder man entwickelt noch ne platine wo man den esp aufsetzt und nen relais integriert ist. Dann könnte man natürlich noch kühlen heizen schalten könnte usw.

**Reply by ap20132022** [reply]
Vor zwei Jahren hab ich ein Shelly verwendet. Es funktioniert auch ganz gut. Problem ist, dass mit Shelly kannst du schwer SG Funktion realisieren. Ich hab damals nur für Aktivierung Heizen verwendet. 
Und dafür braucht man 4 Shelly, welches auch dann gut 50€ kostet. 
Hab ein 4er Relais gekauft. Es kostet bei Alie unter 10€. 
Klar, man kann auch ein 2. ESP mit integriertem Relais verwendet, muss man wieder mehr Geräte, Spannungsversorgung, Geräte in Z.B Home Assistent hinzufüge. 
Ich bin für "ein Gerät". Und bin fest überzeugt, dass es möglich ist und Mehrwert ist.

**Reply by Trunks1982** [reply]
Joa.Aber ich glaube ohne Relais keine Chance.Hier im Forum hatte schon einer ne Platine entwickelt.Normalerweise müsste er das hinbekommen.

**Reply by Trunks1982** [reply]
https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/discussions/35

**Reply by ap20132022** [reply]
Klar, braucht man Relais dafür, weil ESP nur 3 Volt schaltet. 
Auch bei Shelly ist nur Shelly 1 (?) zu verwenden. 
Bei Shelly muss man noch mit 230v Spannungsversorgung aufpassen. 
Ich fand, dass es ESP mit Relais sogar einfach und sicherer. Man kann zur Not sogar Signal negieren.

**Reply by Trunks1982** [reply]
Das stimmt geht nur mit nem Shelly 1.Also muss ne Platine her🤣

**Comment by Roadblock1987** [comment]
Moin,

kurze Anfrage, passt so halb ins Thema ... naja irgendwie zu gut als ein neues aufzumachen...

Wollte in HA ne Automatisierung ... wenn PV genug liefert, dann WW der WP auf ca. 55°C .. leider scheint die AUswahl begrenz auf 51°C.

Wie kann ich das ändern? Übersehe ich was?

Grüße

**Reply by wrfz** [reply]
Hast zwei Möglichkeiten:
1) Kannst aus dem Kombo ein Number-Feld machen. Hast dann die freie Wahl.
    Die Eingabe wird dann aber (meiner Meinung nach) nicht mehr so angenehm. 
    Das machst du, indem du in deinem yaml bei dem WW das hier entfernst: 
    "**type: select**" oder dadurch ersetzt: "**type: number**"

2) Behälst das Kombo, fügst aber zusätzliche Werte hinzu. 
    Siehe dazu `Raumsoll 1` in diesem dieses Beispiel:
    https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/blob/main/examples/options.yaml
    Dort werden benutzerdefinierte Temperatur-Werte definiert.

**Reply by Roadblock1987** [reply]
Hi, danke.

Leider bin ich in dem Ganzen noch nicht tief drin. Ich checke die Grundlagen, aber irgendwie nicht in welcher YAML. ich etwas zu verändern habe. Von meiner Logik her sollte das doch auf dem ESPsein und nicht in der config von HA oder?

Ich gehe davon aus, dass ich das ESPHome Add-On benötige? Bkomm eich den fertigen Rotex ESP dann nur per Kabel da rein? 

GErn auch nen Link um mich tiefer darein zu lesen, statt mir vorsagen zu lassen ;)

Danke für die Hilfe...

Grüße

**Reply by Roadblock1987** [reply]
Moin, mit der neuen Anleitung, etwas rumprobieren und nachdenken hat’s geklappt. Danke
